src/states/elder/mod.rs

```diff
@@ -679,61 +646,18 @@ impl Elder {
 
     // If this returns an error, the peer will be dropped.
     fn handle_bootstrap_request(&mut self, pub_id: PublicId) -> Result<(), RoutingError> {
-        let conn_info = match self.peer_map().get_connection_info(&pub_id) {
-            Some(conn_info) => conn_info.clone(),
-            None => {
-                log_or_panic!(
-                    LogLevel::Error,
-                    "Not connected to the sender of BootstrapRequest."
-                );
-                return Err(RoutingError::UnknownConnection(pub_id));
-            }
-        };
-
-        let (peer_type, client_ip) = match conn_info {
-            ConnectionInfo::Client { peer_addr } => ("client", Some(peer_addr.ip())),
-            ConnectionInfo::Node { .. } => ("node", None),
+        if !self.peer_map.has(&pub_id) {
+            log_or_panic!(
+                LogLevel::Error,
+                "Not connected to the sender of BootstrapRequest."
```

Comment:
.chain
            .valid_peers()
            .neighbour_elders()
            .into_iter()
            .filter(|pub_id| !self.peer_map.has(pub_id))
            .filter(|pub_id| {
            .copied()
                self.peer_mgr.get_peer(pub_id).is_none() && *pub_id != self.full_id.public_id()
            })

Reply:
Currently "connected" / "not connected" pretty much always means "have connection info" / "not have connection info". That said, I'd almost prefer to remove this check altogether because it cannot ever fail unless there is a bug in quic-p2p.